### PR TITLE
Fix issue with order by expressions for non root paths

### PR DESCRIPTION
**Pull request by timowest**
based on https://github.com/querydsl/querydsl/pull/1276
backport https://github.com/querydsl/querydsl/pull/1278

### CODE REVIEW

**Comment by Shredder121** [comment]
Well, in my opinion it looks straightforward, I finished reviewing.
The only thing we *could* enhance is using UnderscoreTemplates, instead of [`oldPath.toString().replace('.', '_')`](https://github.com/querydsl/querydsl/pull/1277/files#diff-5ea4f95c4cb3ea30a76656d43e421599R148).

**Comment by Shredder121** [comment]
This master PR just needs to be rebased, but in my opinion they can be merged.

File: querydsl-jpa/src/main/java/com/querydsl/jpa/JPAQueryMixin.java [hunk 1]
```diff
@@ -145,7 +145,8 @@ private boolean isEntityPath(Path<?> path) {
             Path<?> parent = shorten(metadata.getParent(), paths);
             Path<T> oldPath = new PathImpl<T>(path.getType(),
                     new PathMetadata(parent, metadata.getElement(), metadata.getPathType()));
-            Path<T> newPath = new PathImpl<T>(type, oldPath.toString().replace('.', '_'));
+            Path<T> newPath = new PathImpl<T>(type, ExpressionUtils.createRootVariable(oldPath));
```
Comment (Shredder121) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/JPAQueryMixin.java]: We could use `ExpressionsUtils.path` for the master PR.

Comment (timowest) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/JPAQueryMixin.java]: that's not available, so maybe do the change here https://github.com/querydsl/querydsl/pull/1285 ?

**Pull request merged**